### ISSUE TITLE
New API to get the color format used to create the colord instance

### ISSUE THREAD

**Issue by AnsonH** [omgovich/colord#43]
I recently migrated from [tinycolor2](https://github.com/bgrins/TinyColor) after discovering this awesome library 😄. However, your library is currently missing a similar API to tinycolor2's [`getFormat` method](https://github.com/bgrins/TinyColor#getformat) , which returns the color format used to create the tinycolor instance. I found this API extremely useful when I try to parse user-inputted color code strings.

It would be great if there is a similar API in your library. For example:
```js
var color = colord("#ff0000");
color.getFormat();  // "hex"

color = colord({ r: 139, g: 200, b: 208, a: 1 });
color.getFormat();  // "rgba"
```

Thank you very much!

**Comment by omgovich** [comment]
Hi! It feels like `toString(format?: string)` is not the best way for TS applications.

**Comment by dbismut** [comment]
I think in this case it would be `string`? Or are you questioning naming that function `toString` as it conflicts with native Object.prototype.toString`?

**Comment by omgovich** [comment]
Oh, sorry, my bad 😅 I was thinking about different API when was writing my comment =)
It just feels that having `toString(format)` will force me to create `toObject(format)` which would be hard to type in TS.

**Comment by omgovich** [comment]
Anyway. Having `getFormat` and `toString` should be a big deal. The only thing that bothers me here is the bundle size changes, but I'll try to make the new methods as small as possible.

**Comment by dbismut** [comment]
Haha so that's a case of typings over feature I guess ^^ That's ok no worries I can sort this out, I'll ignore the typings on my side. Thanks!

**Comment by omgovich** [comment]
I'll try to add `getFormat` and `toString` in a few days. Will let you know once I have the first results.

**Comment by omgovich** [comment]
Oof. `getFormat()` appears to be a big deal for me.

Tinycolor2 is mutable and it keeps the same instance when a developer changes a color.

```js
const color1 = tinycolor("#000");
const color2 = color.lighten(10)
console.log(color1 === color2); // true
```

My library is immutable and it creates a new instance on each color change.

In the method below I convert a color to HSL and use it to create a new Colord instance.
```ts
public lighten(amount = 0.1): Colord {
  return colord(lighten(this.rgba, amount));
}
```

It makes it super hard to get the input format after a transformation.

```js
const color1 = colord("#000");
const color2 = color1.lighten(0.1)
console.log(color1 === color2); // false
// 😞
console.log(color1.getFomat()); // 'hex'
console.log(color2.getFomat()); // 'hsl'
```

**Comment by dbismut** [comment]
@omgovich In my specific use case I actually only need `getFormat` to keep track of the initial input.
So actually even this would work for me:

```js
import { getFormat } from 'colord'

const format = getFormat('#000') // hex
const format = getFormat('#ff000033') // hex8
const format = getFormat({ r: 0, g: 0, b: 255 }) // rgb
const format = getFormat({ r: 0, g: 0, b: 255, a: 1 }) // rgba (not sure about that one though)
// etc
```

**Comment by omgovich** [comment]
Yeah, it has to be a simple update of the `parse` I guess.

BTW: Do you think we need to work "hex8" or "rgba". From the library's point of view, I think it makes sense to return a color model of the input (hex, rgb, hsl, lab, lch, ...).

**Comment by dbismut** [comment]
In leva we show the alpha scale of the react-colorful only when the user input has alpha. But if there's another way to check if alpha is part of the user input, then sure, we don't need hex8 or rgba.

**Comment by omgovich** [comment]
Got it. One of the possible ways is:

```ts
colord('#aabbcc80').alpha() < 1 // true or false
```

**Comment by dbismut** [comment]
That might not be enough as for me `#f00000ff` has alpha set although being equal to 1. But don't sweat it if that's too much bloat.

**Comment by omgovich** [comment]
Oh, I see... Why don't you always use alpha pickers BTW?

**Comment by omgovich** [comment]
In opposite to tinycolor which has `toHexString` and `toHex8String`, Colord has single `toHex` method which return hex8 if a color's alpha is less than 1.

**Comment by dbismut** [comment]
Should be fine!

**Comment by omgovich** [comment]
Released v1.7 with `getFormat` utility included 🚀
